pom.xml

```diff
@@ -202,6 +188,11 @@
                 <artifactId>activation</artifactId>
                 <version>1.1.1</version>
             </dependency>
+            <dependency>
+                <groupId>javax.persistence</groupId>
+                <artifactId>javax.persistence-api</artifactId>
+                <version>2.2</version>
+            </dependency>
```

Comment:
Why is this and the dependency to `org.apache.lucene` needed? So far as I know and I see there are no issues or errors related to this two dependencies in supported Java 11 TLS visible. If there are issues in Java 17 or a more recent version - which all of them currently not supported and only Java 17 support is on its way -  then this issues should be fixed when this Java versions are supported and not before. Maybe this issues are already fixed when this Java version are supported in this software.

Reply:
The commonly accepted rule is to name all dependencies which are used (instead of to expect that they are included indirectly). `mvn dependency:analyze` reports the missing ones. This PR adds some of them.

Reply:
Citing @thomaslow: "The recommended strategy of the official [maven-dependency-plugin](https://maven.apache.org/plugins/maven-dependency-plugin/analyze-mojo.html) is that all artifacts whose classes are explicitly imported in source code should also be listed as dependencies in the pom file, even if they are included transitively via other artifacts."

Citing @solth: "I understand that both approaches have their advantages (streamlining pom files vs. explicit dependencies for directly used packages and classes) and would agree with you that we should stick with keeping those dependencies in the pom file, especially when this is an officially recommended strategy."

Reply:
Thank you for the clarification. It would be nice if this kind of message would be included in some way (pull request description, additional note / message in the pull request, ...) so asking why different changes without any connection to the pull request description or something like that was made.